todomvc-app-react/src/sam/TodoApp.js

```diff
@@ -0,0 +1,218 @@
+import React, { useState } from 'react';
+import '../../node_modules/todomvc-common/base.css'
+import '../../node_modules/todomvc-common/base.css'
+
+const clone = (state) => {
+    const comps = state.__components
+    delete state.__components
+    const cln = JSON.parse(JSON.stringify(state))
+    if (comps) {
+      cln.__components = []
+      if (comps.length > 0) {
+        comps.forEach((c) => {
+          delete c.parent
+          cln.__components.push(Object.assign(clone(c), { parent: cln }))
+        })
+      }
+    }
+    return cln
+  }
+
+export let TodoApp = () => {
+    return (<div/>)
+}
```

Comment:
?

Reply:
Just to be safe if someone is trying to use the component before the factory has initialized it.